client/searchResults.jsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+
+import HotelDetails from './hotelDetails';
+import NavBar from './navBar';
+import Loader from './loader';
+export default class SearchResults extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      cityName: this.props.cityName,
+      ratingFilter: this.props.ratingFilter,
+      selectedHotel: '',
+      hotelList: this.props.hotelList
+    };
+    this.handleUserSelection = this.handleUserSelection.bind(this);
+
+  }
+
+  handleUserSelection(e) {
+    e.preventDefault();
+    const hotelList = this.state.hotelList;
+    let selectedIndex = null;
+    for (let i = 0; i < hotelList.length; i++) {
+      if (hotelList[i].name === e.target.textContent) {
+        selectedIndex = i;
+        break;
+      }
+    }
+    this.setState({ selectedHotel: e.target.textContent, selectedIndex: selectedIndex });
+  }
+
+  componentDidMount() {
+    this.setState({ isLoading: true });
+    fetch(`/api/search/${this.state.cityName}`)
+      .then(response => {
+        return response.json();
+      })
+      .then(data => {
+        const cityId = data.suggestions[0].entities[0].destinationId;
+        fetch(`/api/search/list/${cityId}/${this.state.ratingFilter}`).then(response => response.json()).then(data => {
+          const hotelList = data.data.body.searchResults.results;
```

Comment:
```
data.data.body.searchResults.results
```

This is really ugly. Your server should be more polite and just send the results array.